src/Raven.Server/Documents/PeriodicBackup/Restore/Sharding/SingleShardRestoreBackupTask.cs

```diff
@@ -27,42 +26,85 @@ internal class SingleShardRestoreBackupTask : RestoreBackupTask
 
             DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Aviv, DevelopmentHelper.Severity.Major, "test encrypted backup & restore");
             HasEncryptionKey = false;
+
+            _shardNumber = ShardHelper.GetShardNumber(DatabaseName);
+        }
+        
+        protected override Task OnBeforeRestoreAsync()
+        {
+            CreateDocumentDatabase();
             
-            _nonShardedDbName = ShardHelper.ToDatabaseName(DatabaseName);
+            var topology = RestoreSettings.DatabaseRecord.Sharding.Shards[_shardNumber];
+            Database.SetIds(topology, RestoreSettings.DatabaseRecord.Sharding.DatabaseId);
+
+            return Task.CompletedTask;
         }
 
-        protected override RavenConfiguration CreateDatabaseConfiguration()
+        protected override Task OnAfterRestoreAsync()
         {
-            return DatabasesLandlord.CreateDatabaseConfiguration(ServerStore, DatabaseName, RestoreSettings.DatabaseRecord.Settings);
+            SmugglerBase.EnsureProcessed(Result, skipped: false);
+            Progress.Invoke(Result.Progress);
+            return Task.CompletedTask;
         }
 
         protected override DatabaseRecord GetDatabaseRecord()
         {
             using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext serverContext))
             using (serverContext.OpenReadTransaction())
             {
-                var databaseRecord = ServerStore.Cluster.ReadDatabase(serverContext, _nonShardedDbName);
+                var databaseRecord = ServerStore.Cluster.ReadDatabase(serverContext, ShardHelper.ToDatabaseName(DatabaseName));
                 Debug.Assert(databaseRecord != null);
 
                 return databaseRecord;
             }
         }
 
-        protected override Task OnBeforeRestoreAsync()
+        protected override RavenConfiguration CreateDatabaseConfiguration()
         {
-            CreateDocumentDatabase();
-            return Task.CompletedTask;
+            return DatabasesLandlord.CreateDatabaseConfiguration(ServerStore, DatabaseName, RestoreSettings.DatabaseRecord.Settings);
         }
 
-        protected override void OnAfterRestore()
+        protected override async Task ImportLastBackupFileAsync(DocumentDatabase database, DatabaseDestination destination, JsonOperationContext context,
+            DatabaseSmugglerOptionsServerSide options, DatabaseRecord databaseRecord, string lastFilePath)
         {
-            SmugglerBase.EnsureProcessed(Result, skipped: false);
-            Progress.Invoke(Result.Progress);
-        }
+            if (_shardNumber > 0)
+                options.OperateOnTypes &= ~DatabaseItemType.Subscriptions;
```

Comment:
Add a sharding todo here, and create an issue to handle (or at least to discuss) having the most up-to-date database record

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-19202/Sharding-Backup-Restore-consider-using-the-most-up-to-date-database-record